pkg/server/server.go

```diff
@@ -181,13 +181,13 @@ func (s *ProxyServer) getBackend(reqHost string) (Backend, error) {
 
 func (s *ProxyServer) addBackend(agentID string, conn agent.AgentService_ConnectServer) (backend Backend) {
 	for i := 0; i < len(s.BackendManagers); i++ {
+		agentIdentifiers, err := getAgentIdentifiers(conn)
```

Comment:
The default case is "random" across all backends without concern to agent identifiers. As such it should not care if there are any agent identifiers. So I don't think we can early out here. I think we do have to duplicate this check in both DestHostBackendManager and DefaultRouteBackendManager.

Reply:
should be addressed!